File: adapter/src/main/java/com/github/mrmitew/bodylog/adapter/common/model/StateError.java
package com.github.mrmitew.bodylog.adapter.common.model;

public class StateError extends Throwable {
    public static class Empty extends StateError {
        public static final Empty INSTANCE = new Empty();
        private Empty() {
        }

        @Override
        public String toString() {
            return "Empty{}";
        }
    }

    @Override
    public String toString() {
        return "StateError{}";
    }
}

File: adapter/src/main/java/com/github/mrmitew/bodylog/adapter/common/model/StateError.kt
package com.github.mrmitew.bodylog.adapter.common.model

sealed class StateError : Throwable() {
    class Empty private constructor() : StateError() {
        override fun toString(): String {
            return "Empty{}"
        }

        companion object {
            val INSTANCE = Empty()
        }
    }

    override fun toString(): String {
        return "StateError{}"
    }
}